hnix-store-remote/src/System/Nix/Store/Remote.hs

```diff
@@ -222,7 +218,7 @@ queryPathInfoUncached path = do
   deriverPath <- sockGetPathMay
 
   narHashText <- Data.Text.Encoding.decodeUtf8 <$> sockGetStr
-  let narHash = case System.Nix.Hash.decodeBase32 @'System.Nix.Hash.SHA256 narHashText of
+  let narHash = case (System.Nix.Hash.decodeBase Base32 :: Text -> Either String (Digest a)) @'System.Nix.Hash.SHA256 narHashText of
```

Comment:
```suggestion
  let narHash = case System.Nix.Hash.decodeBase @'System.Nix.Hash.SHA256 Base32 narHashText of
```

Reply:
Hmm....

I thought the visible type application goes right before the argument it casts type onto.

Now I understood your comment about type application being the first argument.